test.disabled/ejabberd_tests/run_common_test.erl

```diff
@@ -453,3 +456,67 @@ travis_fold(Description, Fun) ->
 import_code_paths(FromNode) when is_atom(FromNode) ->
     Paths = rpc:call(FromNode, code, get_path, []),
     code:add_paths(Paths).
+
+%% covecov.json cover format
+%% ------------------------------------------------------------------
+
+%% Writes codecov.json into root of the repo directory
+%%
+%% Format:
+%%
+%% {"coverage": {
+%%     "src/mod_mam.erl": [0, 1, 4, null]
+%% }}
+%%
+%% Where:
+%%
+%% - 0 - zero hits
+%% - null - lines, that are not executed (comments, patterns...)
+%% - 1 - called once
+%% - 4 - called 4 times
+export_codecov_json() ->
+    Modules = cover:imported_modules(),
+    %% Result is a flat map of `{{Module, Line}, CallTimes}'
+    {result, Result, _} = cover:analyse(Modules, calls, line),
+    Mod2Data = lists:foldl(fun add_cover_line_into_array/2, #{}, Result),
+    JSON = maps:fold(fun format_array_to_list/3, [], Mod2Data),
+    Binary = jiffy:encode(#{<<"coverage">> => {JSON}}),
+    file:write_file(?ROOT_DIR ++ "/codecov.json", Binary).
+
+add_cover_line_into_array({{Module, Line}, CallTimes}, Acc) ->
+    %% Set missing lines to null
+    CallsPerLineArray = maps:get(Module, Acc, array:new({default, null})),
+    Acc#{Module => array:set(Line, CallTimes, CallsPerLineArray)}.
+
+format_array_to_list(Module, CallsPerLineArray, Acc) ->
+    ListOfCallTimes = array:to_list(CallsPerLineArray),
+    BinPath = list_to_binary(get_source_path(Module)),
+    [{BinPath, ListOfCallTimes}|Acc].
+
+%% We assume that all mongooseim modules are loaded on this node
+get_source_path(Module) when is_atom(Module) ->
+    try
+        AbsPath = proplists:get_value(source, Module:module_info(compile)),
+        string:prefix(AbsPath, get_repo_dir())
```

Comment:
AFAIK this function was introduced in Erlang 20, so paths would be messed up in other builds.